voidcrew/code/game/objects/items/devices/scanners.dm

```diff
@@ -26,3 +26,69 @@
 	var/response = SEND_SIGNAL(M, COMSIG_NANITE_SCAN, user, TRUE)
 	if(!response)
 		to_chat(user, "<span class='info'>No nanites detected in the subject.</span>")
+
+/obj/item/bio_scanner
+	name = "Biological Scanner"
+	desc = "A small device that performs a detailed scan of various living or descesed beings and uploads the collected data to a connected research server."
+	icon = 'icons/obj/device.dmi'
+	icon_state = "forensicnew" // should be temp
+	w_class = WEIGHT_CLASS_SMALL
+	item_state = "electronic" // should be temp
+	lefthand_file = 'icons/mob/inhands/misc/devices_lefthand.dmi' // should be temp
+	righthand_file = 'icons/mob/inhands/misc/devices_righthand.dmi' // should be temp
+	flags_1 = CONDUCT_1
+	item_flags = NOBLUDGEON
+	slot_flags = ITEM_SLOT_BELT
+	throwforce = 3
+	w_class = WEIGHT_CLASS_TINY
+	throw_speed = 3
+	throw_range = 7
+	var/datum/techweb/stored_research
+	/// the actual reward
+	var/value = 0
+	/// base reward before bonus, todo: make this a var in /mob/
+	var/reward = 1500
+
+/obj/item/bio_scanner/attack(mob/living/attacked_mob, mob/living/user)
+	if(!istype(attacked_mob))
+		return //Scanning
+	if(!stored_research)
+		playsound(loc, 'sound/effects/zzzt.ogg', 25, TRUE)
+		say("Cannot scan. Not connected to a research server! Tap server with device to link.")
+	else if(stored_research.scanned_mobs.Find(attacked_mob.name))
+		playsound(loc, 'sound/effects/zzzt.ogg', 25, TRUE)
+		say("Scan failed. Already scanned!")
+	else if(do_after(user, 10))
+		stored_research.scanned_mobs.Add(attacked_mob.name)
+		getvalue(attacked_mob)
+		stored_research.add_point_list(list(TECHWEB_POINT_TYPE_GENERIC = value))
+		say("Sucessfully scanned. [value] points added to database.")
+		playsound(loc, 'sound/effects/adminhelp.ogg', 25, TRUE) //lul
+	else
+		playsound(src, 'sound/machines/buzz-sigh.ogg', 20)
+		say("Must be stationary to scan!")
+		return
+
+/obj/item/bio_scanner/AltClick(mob/user)
+	stored_research = null
+	say("Cleared linked techweb!")
+
+/obj/item/bio_scanner/proc/getvalue(mob/living/M) // Copy pasted from experimental_disection.dm I'll be working on a PR soon to give /mob a new var that hold its point reward, so no more of this bs
+	var/mob/living/target = M
```

Comment:
you marked the reviews as resolved here but didnt actualy resolve them :(

Reply:
The getvalue proc is just copy pasted from experimental_dissection and is temporary, once this PR is merged I'm making another that gives mob/ a new var that's a default research value, and then this code and dissections code will be updated to use that value, as opposed to that proc. tbh I'm not even sure what some of what you pointed out even does.